File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int answer = 0; 
        int currMin = prices[0]; 
        for(int i = 1; i < prices.size(); i++){
            answer = max(answer, prices[i] - currMin); 
            currMin = min(currMin, prices[i]); 
        }

        return answer; 
    }
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int maxPrice = 0; 
        int currPrice = prices[0]; 
        for(int i = 1; i < prices.length; i++){
            maxPrice = Math.max(maxPrice, prices[i] - currPrice); 
            currPrice = Math.min(currPrice, prices[i]); 
        }
        return maxPrice; 
    }
}